demo/async-read-file.js

```diff
@@ -1,18 +1,30 @@
 // Run this with
 //
-//   traceur --experimental async-read-file.js file-to-read
+//   traceur --async-functions=true async-read-file.js
 
-function lineCount(file) {
-  var data;
-  try {
-    // TODO: make await an expression.
-    await data = require('fs').readFile(file, 'utf8');
-    console.log(`${file} has ${data.split('\n').length} lines`);
-  } catch (ex) {
-    console.error(ex);
-  }
+function asyncReadFile(file) {
+  return new Promise((resolve) => {
+    require('fs').readFile(file, 'utf8', (err, data) => {
+      if (err)
+        throw err;
+      resolve(data);
+    });
+  });
 }
 
-var fileName = process.argv[2];
-lineCount(fileName);
-console.log(`Loading ${fileName}...`);
+(async function() {
+  // Set filename to demo/async-read-file.js (this file).
+  let filename = __filename;
+
+  // Asynchronously starts the read process.
+  let promise = asyncReadFile(filename);
```

Comment:
This is backwards. No need to do this extra step. Just do:

```js
console.log(`Loading ${filename}...`);
var file = await asyncReadFile(filename);
```

I also feel like the try/catch is an important part of the demo.

```js
console.log(`Loading ${filename}...`);
var data;
try {
  data = await asyncReadFile(filename);
  console.log(`${filename} has ${data.split('\n').length} lines`);
} catch (ex) {
  console.error(ex);
}
```